From 1f6cec0f4bfacf913ea2e0a9a44a37011b31143a Mon Sep 17 00:00:00 2001
From: Danny Nold <dannynold@freescale.com>
Date: Tue, 4 Jan 2011 09:30:37 -0600
Subject: [PATCH] ENGR00137580 - EPDC fb: Endless collision due to bug in collision handling

- When comparing an update with those it has collided with, do not
resubmit the update if the counter is greater than OR equal to that
of the collided-with updates.

Signed-off-by: Danny Nold <dannynold@freescale.com>
---
 drivers/video/mxc/mxc_epdc_fb.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/mxc/mxc_epdc_fb.c b/drivers/video/mxc/mxc_epdc_fb.c
index 337f763..0a6981a 100644
--- a/drivers/video/mxc/mxc_epdc_fb.c
+++ b/drivers/video/mxc/mxc_epdc_fb.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2010 Freescale Semiconductor, Inc.
+ * Copyright (C) 2010-2011 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -2523,9 +2523,9 @@ static irqreturn_t mxc_epdc_irq_handler(int irq, void *dev_id)
 				if (!(temp_mask & 0x1))
 					continue;
 
-				if (fb_data->lut_update_order[lut] >
+				if (fb_data->lut_update_order[lut] >=
 					fb_data->cur_update->update_order) {
-					dev_dbg(fb_data->dev, "Ignoring collision with new update.\n");
+					dev_dbg(fb_data->dev, "Ignoring collision with newer update.\n");
 					ignore_collision = true;
 					break;
 				}
-- 
1.5.4.4

